src/resources.cpp

```diff
@@ -1,3 +1,7 @@
 #include "kaacore/resources.h"
 
-namespace kaacore {} // namespace kaacore
+namespace kaacore {
+
+Resource::~Resource() {}
```

Comment:
can we put:
```
    if (this->is_initialized) {
        this->_uninitialize();
    }
```
here, co it does not repeat in desended destructors?

Reply:
nope, you are now allowed to call pure virtual function from destructor